### PR TITLE
Rework #17486

### PR DESCRIPTION
**Pull request by t-nelson**
#### Problem

#17486 introduced several issues
1) Easy to misuse and require a default wallet to exist for commands that didn't need it. This was the case for at least `solana validators`
2) Assumed the path was a filepath, when it could be a bunch of other signer types which don't exist in a filesystem
3) Made the assumption that "keypair" was always the arg name, which it is not for `spl-token` CLI

#### Summary of Changes

* Revert #17486
* Do an existence only for filepath `SignerSources`

Review hints:
* Ignore the first commit
* Do a better job than I did on the original :sweat_smile:

### CODE REVIEW

File: clap-utils/src/keypair.rs [hunk 1]
```diff
@@ -215,6 +195,23 @@ impl std::fmt::Debug for SignerSourceKind {
...
 }
 
+impl SignerSourceKind {
+    pub fn new_filepath_checked(path: String) -> Result<Self, SignerSourceError> {
+        std::fs::metadata(&path)
+            .map_err(|_| {
+                std::io::Error::new(
+                    std::io::ErrorKind::Other,
+                    format!(
+                    "No default signer found, run \"solana-keygen new -o {}\" to create a new one",
```
Comment (jackcmay) [inline clap-utils/src/keypair.rs:205]: This message no longer only applies to the "default", but any singer source.

**Comment by jackcmay** [comment]
There was a suggestion that we just force cli users to setup a default keypair because they will eventually need it anyway and might as well have them address it from the start.

File: clap-utils/src/keypair.rs (hunk 1 above)
Comment (t-nelson) [reply clap-utils/src/keypair.rs:205]: Good point! Not sure if I should generalize the message or revisit my first pass that deferred the `stat()` call until the first query :thinking:

Comment (t-nelson) [reply clap-utils/src/keypair.rs]: I went with the latter

**Review by jackcmay — DISMISSED** [review]

**Review by CriesofCarrots — DISMISSED** [review]

**Pull request merged**